ui/lib/sync/addon/components/secrets/page/destinations.hbs

```diff
@@ -12,4 +12,114 @@
+      data-test-filter="type"
+    />
+    <SearchSelect
+      @options={{this.destinationNames}}
+      @objectKeys={{array "id" "name"}}
+      @passObject={{true}}
+      @selectLimit={{1}}
+      @disallowNewItems={{true}}
+      @placeholder="Filter by name"
+      @inputValue={{if @nameFilter (array @nameFilter)}}
+      @onChange={{fn this.onFilterChange "name"}}
+      class="is-marginless has-left-padding-s"
+      data-test-filter="name"
+    />
+  </ToolbarFilters>
+  <ToolbarActions>
+    <ToolbarLink @route="secrets.destinations.create" @type="add" data-test-create-destination>
+      Create new destination
+    </ToolbarLink>
+  </ToolbarActions>
+</Toolbar>
+
+{{#if @destinations.meta.filteredTotal}}
+  <div class="has-bottom-margin-s">
+    {{#each @destinations as |destination index|}}
+      <ListItem
+        @linkPrefix={{this.mountPoint}}
+        @linkParams={{array "secrets.destinations.destination.details" destination.type destination.name}}
+        as |Item|
+      >
+        <Item.content>
+          <div>
+            <Icon @name={{destination.icon}} data-test-destination-icon={{index}} />
+            <span data-test-destination-name={{index}}>
+              {{destination.name}}
+            </span>
+          </div>
+          <code class="has-text-grey is-size-8" data-test-destination-type={{index}}>
+            {{destination.typeDisplayName}}
+          </code>
+        </Item.content>
+
+        <Item.menu as |Menu|>
+          {{#if destination.destinationPath.isLoading}}
+            <li class="action">
+              <Hds::Button disabled @color="tertiary" @icon="loading" @text="loading" @isIconOnly={{true}} />
+            </li>
+          {{else}}
+            <li>
+              <LinkTo
+                class="has-text-black has-text-weight-semibold"
+                data-test-details
+                @route="secrets.destinations.destination.details"
+                @models={{array destination.type destination.name}}
+                @disabled={{not destination.canRead}}
+              >
+                Details
+              </LinkTo>
+            </li>
+            <li>
+              <LinkTo
+                class="has-text-black has-text-weight-semibold"
+                data-test-edit
+                @route="secrets.destinations.destination.edit"
+                @models={{array destination.type destination.name}}
+                @disabled={{not destination.canEdit}}
+              >
+                Edit
+              </LinkTo>
+            </li>
+            <li>
+              <Menu.Message
+                data-test-delete
+                @id={{concat "delete-" destination.id}}
+                @triggerText="Delete"
+                @title="Are you sure?"
+                @message="The destination will be permanently deleted and all the secrets will be unsynced. This cannot be undone."
+                @onConfirm={{fn this.onDelete destination}}
+              />
```

Comment:
This should be wrapped in a capabilities check but there is a separate ticket to address permissions.